src/controller/handler/issue.credential.ts

```diff
@@ -83,28 +95,36 @@ export class IssueCredential implements IAgentResponseHandler {
             return res.data;
         }
 
-        if (body.role === 'holder' && body.state === 'request_sent') {
-            const action = 'store';
+        // Not sure why, but sometimes the role for issuer comes back as undefined
+        if ((body.role === 'issuer' || body.role === undefined) && body.state === 'request_received') {
+            if (body.auto_issue === true) {
+                Logger.info(`Not requesting issuer to issue credential, because auto_issue is true`);
+                return;
+            }
+            const action = 'issue';
             const templatedCacheKey = `${agentId}-${body.role}-${body.credential_exchange_id}`;
             await this.checkPolicyForAction(action, templatedCacheKey);
             await readPermission(action, templatedCacheKey);
 
-            await ProtocolUtility.delay(2000);
```

Comment:
Im a little nervous deleting this delay.  I tried removing this when I was finishing up the governance changes and tests would fail without it.  I think we need to leave it.

Reply:
So the key is that we used to process this when the state was 'request_sent' which was too early. Now that we're doing this on state 'credential_received', everything is ready to /store so no need to delay.

Reply:
I got all tests passing with these changes here: https://github.com/kiva/aries-guardianship-agency/pull/93